var/spack/repos/builtin/packages/intel-mkl/package.py

```diff
@@ -68,30 +77,35 @@ def blas_libs(self):
             mkl_integer = ['libmkl_intel_lp64']
 
         mkl_threading = ['libmkl_sequential']
+        omp_threading = []
 
         if '+openmp' in spec:
             if '%intel' in spec:
-                mkl_threading = ['libmkl_intel_thread', 'libiomp5']
+                mkl_threading = ['libmkl_intel_thread']
+                omp_threading = ['libiomp5']
             else:
                 mkl_threading = ['libmkl_gnu_thread']
 
         # TODO: TBB threading: ['libmkl_tbb_thread', 'libtbb', 'libstdc++']
 
-        mkl_root = join_path(prefix.lib, 'intel64')
+        mkl_root = prefix.compilers_and_libraries.linux.mkl.lib.intel64
```

Comment:
you have this variable defined 3 times (and another 2 times in `intel-parallel-studio`). Maybe instead introduce a (documented) method to return `mkl_root` and `omp_root` for a given prefix and use it below?